### DISCUSSION TITLE
Programatically click object in GeoJson layer

### DISCUSSION THREAD

**Discussion by hoang-ta** [visgl/deck.gl#5735] [Q&A] [answered]
Hello everyone, is there a way to trigger a click programatically on a specific object (feature) of a GeoJson layer through code not by click interaction with the map? I am trying to perform a click based on a feature's property

**Comment by padawannn** [comment, marked as answer]
When you talk about trigger a click programmatically, what result do you hope to obtain exactly? highlight the feature, show a popup, center the map...?

**Reply by hoang-ta** [reply]
FYI, i am using React to manage the GeoJson data. Currently, I allow user to change feature color when clicking on it based on the `pickInfo`

`onClick(pickInfo, event) => {'logic code to change color base on the pickInfo'}`

Now I receive message from outside to change the color, the message contains only the ID of the feature. Currently, I change the color by manipulating the GeoJson data based on the ID. 

So i  wonder if  there is a different way to do it (like trigger an action of DeckGL without interacting with it).

**Reply by padawannn** [reply]
Instead of changing the color by manipulating the GeoJson, you should use the `getFillColor` layer property and `updateTriggers`. Check this https://deck.gl/docs/developer-guide/performance#use-updatetriggers

In the example, ya can see how to change the layer radius based on `selectYear` without changing the data. In your case, the  `selectYear` would be the message from outside
